contracts/farm-bot.sol

```diff
@@ -137,11 +146,29 @@ contract FarmBot is Owned, FarmbotERC20 {
     }
 
     // convenience method for anyone considering calling claimRewards (who may want to compare bounty to gas cost)
-    function previewBounty() external view returns (TokenAmount[] memory) {
-        uint _leftoverBalance = rewardsToken.balanceOf(address(this));
-        uint _interestEarned = stakingRewards.earned(address(this));
-        TokenAmount[] memory _rewardsTokenBalances = new TokenAmount[](1);
-        _rewardsTokenBalances[0] = TokenAmount(rewardsToken, _interestEarned + _leftoverBalance);
+    // Annoyingly, the MoolaStakingRewards.earnedExternal method is not declared as a view, so we cannot declare this
+    // method as a view itself.
+    function previewBounty() external returns (TokenAmount[] memory) {
+	uint[] memory _leftoverBalances = new uint[](rewardsTokens.length);
+	for (uint i=0; i < rewardsTokens.length; i++) {
+	    _leftoverBalances[i] = rewardsTokens[i].balanceOf(address(this));
+	}
+
+	// The MoolaStakingRewards contract treats the "native" reward token as fundamentally
+	// different than the "external" ones, so we have to query the earned balance separately
+	uint[] memory _interestEarned = new uint[](rewardsTokens.length);
+	_interestEarned[0] = stakingRewards.earned(address(this));
+
+	uint[] memory _externalEarned = stakingRewards.earnedExternal(address(this));
+	for (uint i=0; i < _externalEarned.length; i++) {
+	    _interestEarned[i+1] = _externalEarned[i];
+	}
+
+        TokenAmount[] memory _rewardsTokenBalances = new TokenAmount[](rewardsTokens.length);
+	for (uint i=0; i < rewardsTokens.length; i++) {
+	    _rewardsTokenBalances[i] = TokenAmount(rewardsTokens[i], _interestEarned[i] + _leftoverBalances[i]);
```

Comment:
Seems like there is an important dependency on `rewardsTokens` having a particular order. Can you document this somewhere? Probably in the constructor?

Reply:
Yes, that's definitely true; really important to have this documented clearly.